sw/device/lib/crypto/drivers/hmac.c

```diff
@@ -89,6 +75,8 @@ enum {
  * It returns error if HMAC HWIP becomes idle without firing `hmac_done`
  * interrupt.
  *
+ * TODO(#23191): It might be beneficial to have a timeout value for the polling.
+ *
```

Comment:
Doing a plain revert of the other commits is fine and the approach is good, it's easier to review this.

What is not ideal is that it means to re-introduce this TODO here which got resolved by the workaround. Would it be possible to do a separate commit or even PR (whatever you prefer @andrea-caforio ) to resolve this? You can take inspiration from the reverted commits on how to do this.

Reply:
Thank you @vogelpi for the review and catching this issue. I reinstantiated the polling timeout implementation in a separate commit as to not spam the PR list with code that has already been review before.